### PR TITLE
rgw multisite: make host in zone_endpoints_list configurable

### PR DESCRIPTION
**Pull request by alimaredia**
Also change endpoint to pull_endpoint

Signed-off-by: Ali Maredia <amaredia@redhat.com>

### CODE REVIEW

**Comment by alimaredia** [comment]
Attempts to address https://github.com/ceph/ceph-ansible/issues/5165

**Comment by dsavineau** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by dsavineau** [comment]
jenkins test centos-container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by p37ruh4** [comment]
Any updates on this?

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
@alimaredia could you squash your commits please?

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

File: roles/ceph-handler/templates/restart_rgw_daemon.sh.j2 [hunk 1]
```diff
@@ -11,7 +11,7 @@ else
     RGW_PROTOCOL=http
 fi
 INSTANCES_NAME=({% for i in rgw_instances %}{{ i.instance_name }} {% endfor %})
-RGW_IPS=({% for i in rgw_instances %}{{ i.radosgw_address }} {% endfor %})
+RGW_IPS=({% for i in rgw_instances %}{{ i.rgw_multisite_host }} {% endfor %})
```
Comment (guits) [inline roles/ceph-handler/templates/restart_rgw_daemon.sh.j2:14]: this is a breaking change

File: roles/ceph-defaults/defaults/main.yml [hunk 2]
```diff
@@ -522,13 +522,12 @@ rgw_zone: default
 #rgw_zone_user_display_name: "Zone User"
 #rgw_realm: milkyway # should be set by the user
 #rgw_multisite_proto: "http"
+#rgw_multisite_host: "{{ _radosgw_address if rgw_multisite_proto == 'http' else ansible_fqdn }}"
```
Comment (guits) [inline roles/ceph-defaults/defaults/main.yml:525]: why are you leaving this commented out?

Comment (alimaredia) [reply roles/ceph-defaults/defaults/main.yml:525]: Since the variable shouldn't be set unless `rgw_multisite: true`

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-all_daemons

**Comment by alimaredia** [comment]
jenkins test centos-container-all_daemons

**Comment by alimaredia** [comment]
jenkins test centos-non_container-all_daemons

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by alimaredia** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
@alimaredia is there anything I can do here to help ?

**Comment by guits** [comment]
@alimaredia ?

**Comment by alimaredia** [comment]
Yes, I need help understanding how to fix the issues with the integration tests in `restart_rgw_daemon.sh.j2`. I'm not sure how to apply the changes made in the PR into that file.

Also I looked at the master branch recently, and it looks like a lot of the multisite code has been updated with custom ansible modules. I have no idea how a rebase would work, so that would be worth discussing as well

File: roles/ceph-facts/tasks/set_radosgw_address.yml [hunk 3]
```diff
@@ -59,7 +59,7 @@
...
+    rgw_instances: "{{ rgw_instances|default([]) | union([{'instance_name': 'rgw' + item | string, 'rgw_multisite_proto': rgw_multisite_proto, 'rgw_multisite_host': rgw_multisite_host, 'radosgw_address': rgw_multisite_host, 'radosgw_frontend_port': radosgw_frontend_port | int, 'rgw_realm': rgw_realm | string, 'rgw_zonegroup': rgw_zonegroup | string, 'rgw_zone': rgw_zone | string, rgw_zonemaster: 'rgw_zonemaster, rgw_zonesecondary: 'rgw_zonesecondary', rgw_zonegroupmaster: 'rgw_zonegroupmaster, 'system_access_key': system_access_key, 'system_secret_key': system_secret_key, 'rgw_zone_user': rgw_zone_user, 'rgw_zone_user_display_name': rgw_zone_user_display_name, 'rgw_pull_endpoint': rgw_pull_endpoint }]) }}"
```
Comment (guits) [inline roles/ceph-facts/tasks/set_radosgw_address.yml:62]: @alimaredia why are you removing `+ item|int` from this:

```yaml
'radosgw_frontend_port': radosgw_frontend_port | int + item|int
```

Comment (guits) [inline roles/ceph-facts/tasks/set_radosgw_address.yml:62]: the syntax is invalid at several places:
ie: `rgw_zonemaster: 'rgw_zonemaster`

**Comment by guits** [comment]
@alimaredia I've `git push -f` in this branch: don't forget to `git fetch; git reset --hard origin/wip-rgw-multisite-host-address`

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Pull request closed without merging**